### ISSUE TITLE
HL2 not attaching/working

### ISSUE THREAD

**Issue by Dal1ster** [NVIDIAGameWorks/rtx-remix#58]
**Describe the bug**
- When extracting the contents of the download to their respective places in the games folders, upon starting up the game, it doesnt apply remix to it, and boots like "normal".
- 0.1.1
- NVIDIA GeForce RTX 3060
- 531.61
- Both Games are from steam

**Attach logs!**
- NvRemixBridge.log
- d3d9.log
[NvRemixBridge.log](https://github.com/N
[hl2_d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11226096/hl2_d3d9.log)
VIDIAGameWorks/rtx-remix/files/11226093/NvRemixBridge.log)
[d3d9.log](https://github.com/NVIDIAGameWorks/rtx-remix/files/11226097/d3d9.log)

- The game specific log will be named gamename_d3d9.log

**To Reproduce**
1. Download specified version
2. Extract to hl2 root folder
3. run game
4. see result

**Expected behavior**
I expected it to boot with remix enabled along with the splash saying the shortcut to access settings for it

**Screenshots**

**Comment by sudological** [comment]
It looks like Source games need a slightly different install process, see [this other issue](https://github.com/NVIDIAGameWorks/rtx-remix/issues/25) You don't just dump all the files to the root folder with hl2.exe, you actually need to move d3d9.dll and the .trex folder into the /bin/ folder.

**Comment by SirenGlitch** [comment]
>and the .trex folder

Ah, so thats what i've been missing, someone mentioned the dll in /bin but never .trex. will try that now!

**Comment by Dal1ster** [comment]
So it boots with remix, but then does something very unexpected. After a little bit, the world will start "flashing" a bright white and black (i think) really fast.

**Comment by Dal1ster** [comment]
Thats odd, upon rebooting the game, it now just shows a black sky instaid of the flashing one from before. Video for refrence
[NOTE: I Had to use drive since nivdia share recording file sizes are way too big to upload directly here 
[https://drive.google.com/file/d/1fr4uslqASFrgTU1UC44uOzO09uSNbc62/view?usp=sharing](url)

**Comment by MarkEHenderson** [comment]
Please try running Half Life 2 using these instructions:
```
Half Life 2 recommended installation instructions:
1. Install Half Life 2 and PortalRTX.  Make sure both are fully updated - Portal With RTX just got a bugfix patch as of 2:30 PST today.
2. Copy the contents of "common\PortalRTX\portal_rtx\cfg" to "common\Half-Life 2\hl2\cfg" overwriting existing files
3. Copy the contents of "common\PortalRTX\bin" to "common\Half-Life 2\bin" overwriting the existing files
4. Copy rtx.conf and dxvk.conf from "common\PortalRTX" to "common\Half-Life 2"
5. Extract the contents of Remix 0.1.1 to "common\Half-Life 2\bin" overwriting existing files
6. Launch from steam.
```

**Comment by RoyaleNoir** [comment]
> Please try running Half Life 2 using these instructions:
> 
> ```
> Half Life 2 recommended installation instructions:
> 1. Install Half Life 2 and PortalRTX.  Make sure both are fully updated - Portal With RTX just got a bugfix patch as of 2:30 PST today.
> 2. Copy the contents of "common\PortalRTX\portal_rtx\cfg" to "common\Half-Life 2\hl2\cfg" overwriting existing files
> 3. Copy the contents of "common\PortalRTX\bin" to "common\Half-Life 2\bin" overwriting the existing files
> 4. Copy rtx.conf and dxvk.conf from "common\PortalRTX" to "common\Half-Life 2"
> 5. Extract the contents of Remix 0.1.1 to "common\Half-Life 2\bin" overwriting existing files
> 6. Launch from steam.
> ```

An alternative that doesn't involve overwriting the engine binaries would likely just be to copy the `common\Half-Life 2\hl2` folder into `common\PortalRTX\hl2` and launching it by just running `common\PortalRTX\hl2.exe`

**Comment by SirenGlitch** [comment]
Yes, but that would destroy the PortalRTX installation. You would need to reinstall to fix it

**Comment by RoyaleNoir** [comment]
Would it? The hl2 folder already in portal is just the content used in a regular install of hl2. As long as the vpks are still there Portal should theoretically still run.

**Comment by RoyaleNoir** [comment]
The reason that regular source doesn't work with RTX Remix seems to be that source 2013 is [hardcoded to only go down to dxlevel 80](https://github.com/ValveSoftware/source-sdk-2013/blob/master/sp/src/public/materialsystem/imaterialsystem.h#L73l), which still uses Shader Model 1 shaders. I think RTX Remix is meant to have partial support for dx8 shaders but it doesn't seem to work in this case.

using `bridge.conf` to set the device's reported shader version (`client.shaderVersion`) to 0 just causes source to crash, presumably due to the dxlevel being hardcoded to 80 and above and the [shaders used in dxlevel 80 only falling back to fixed function when the dxlevel is below 80](https://github.com/ValveSoftware/source-sdk-2013/blob/master/sp/src/materialsystem/stdshaders/lightmappedgeneric_dx8.cpp#L204).

**Comment by sudological** [comment]
> The reason that regular source doesn't work with RTX Remix seems to be that source 2013 is [hardcoded to only go down to dxlevel 80](https://github.com/ValveSoftware/source-sdk-2013/blob/master/sp/src/public/materialsystem/imaterialsystem.h#L73l), which still uses Shader Model 1 shaders. I think RTX Remix is meant to have partial support for dx8 shaders but it doesn't seem to work in this case.
> 
> using `bridge.conf` to set the device's reported shader version (`client.shaderVersion`) to 0 just causes source to crash, presumably due to the "dxsupportlevel" also never seeming to go below 80 and the [shaders used in dxlevel 80 only falling back to fixed function when the dxsupportlevel is below 80](https://github.com/ValveSoftware/source-sdk-2013/blob/master/sp/src/materialsystem/stdshaders/lightmappedgeneric_dx8.cpp#L204). There may also be some that don't have fallbacks at all.

Hmm. Perhaps we would get better results with the Source Unpack versions that speedrunners use, as these use older Source Engine versions.

**Comment by NV-LL** [comment]
Hello @Dal1ster! Will you please test this on the latest release and let us know if the issue persists? Thank you!

**Comment by Dal1ster** [comment]
I tried to run the game with the latest version, but nothing opens or happens window wise, and according to steam, immediately closes afterwards @NV-LL

**Comment by NV-LL** [comment]
Hi @Dal1ster & @Slluxx - can you please send logs/dump files if you're still seeing incorrect textures and RTX not working? Also, some screenshots or video of what you see? Thanks!

**Comment by NV-LL** [comment]
Hey @Dal1ster & @Slluxx - checking in again to see if you can provide the info requested above. We'd also like to inform you of our stale ticket policy, specifically that this ticket will be closed as 'went stale' if it does not receive an update in the next 2 weeks. Thank you!

**Comment by Slluxx** [comment]
Sorry, this was the only thing i was interested in and because i replied to the ticket in november 2024 without a response for so long, i uninstalled the rtx env and moved on to other things. But i can assure you it was a fresh install of rtx and hl2 and just wasnt working.
